internal/ackhandler/received_packet_tracker.go

```diff
@@ -143,9 +141,19 @@ func (h *receivedPacketTracker) maybeQueueACK(pn protocol.PacketNumber, rcvTime
 		h.ackAlarm = rcvTime.Add(h.maxAckDelay)
 	}
 
-	// Queue an ACK if there are new missing packets to report.
+	// queue an ACK if there are new missing packets to report
 	if h.hasNewMissingPackets() {
-		h.logger.Debugf("\tQueuing ACK because there's a new missing packet to report.")
+		if h.logger.Debug() {
+			h.logger.Debugf("\tQueuing ACK because there's a new missing packet to report.")
+		}
+		h.ackQueued = true
+	}
+
+	// queue an ACK if the packet was ECN-CE marked
+	if ecn == protocol.ECNCE {
+		if h.logger.Debug() {
```

Comment:
This is likely important for performance. When using L4S, we expect 2 CE-markings per RTT, so we want to avoid calls to `Debugf` when we're not actually outputting logs.

Reply:
This will just be a function call with a few params, no? 

```
// Debugf logs something
func (l *defaultLogger) Debugf(format string, args ...interface{}) {
	if l.logLevel == LogLevelDebug {
		l.logMessage(format, args...)
	}
}
```

Reply:
Right. It's only important when passing arguments to the function, because that would allocate a slice.